common/src/chain/transaction/input.rs

```diff
@@ -23,6 +25,29 @@ impl OutPoint {
     pub fn get_output_index(&self) -> u32 {
         self.index
     }
+
+    pub fn is_coinbase(&self) -> bool {
+        self.id.is_null() && self.index == Self::COINBASE_OUTPOINT_INDEX
+    }
+}
+
+impl PartialOrd for OutPoint {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        self.id.get().partial_cmp(&other.id.get()).and_then(|ordering| match ordering {
+            std::cmp::Ordering::Equal => self.index.partial_cmp(&other.index),
+            _ => Some(ordering),
+        })
+    }
+}
+
+impl Ord for OutPoint {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        let id_ord = self.id.get().cmp(&other.id.get());
+        match id_ord {
+            std::cmp::Ordering::Equal => self.index.cmp(&other.index),
+            _ => id_ord,
+        }
+    }
```

Comment:
How is this different from `derive(Ord, PartialOrd)`?

Reply:
This is differenct because we are comparing the `H256`s instead of `Id<Transaction>`s.
If `OutPoint` as annotated with `derive(Ord, PartialOrd)`, the compiler requires these two traits to be implemented for `Transaction`, for reasons I don't fully understand. This would require `Transaction`'s entire sub-tree of struct to implement the traits. I'm not sure this is what we want.
So I implemented the traits manually to get around the compiler's unreasonable (in my eye) demand.